File: lib/features/forums/forum_tags.dart
String normalizeForumTag(String raw) {
  return raw
      .trim()
      .toLowerCase()
      .replaceAll(RegExp(r'\s+'), '-')
      .replaceAll(RegExp(r'[^a-z0-9\-áéíóúüñ]'), '')
      .replaceAll(RegExp(r'-+'), '-')
      .replaceAll(RegExp(r'^-|-$'), '');
}

List<String> normalizeForumTags(Iterable<String> raw) {
  final seen = <String>{};
  final result = <String>[];
  for (final item in raw) {
    final tag = normalizeForumTag(item);
    if (tag.isEmpty || seen.contains(tag)) continue;
    seen.add(tag);
    result.add(tag);
    if (result.length >= 5) break;
  }
  return result;
}

File: apps/web/lib/forum-tags.ts
/** Normalize and dedupe forum tag slugs (max 5). */
export function normalizeForumTags(tags: string[]) {
  const seen = new Set<string>();
  const out: string[] = [];
  for (const raw of tags) {
    const tag = raw
      .trim()
      .toLowerCase()
      .replace(/[^a-z0-9-]+/g, "-")
      .replace(/^-+|-+$/g, "");
    if (!tag || seen.has(tag)) continue;
    seen.add(tag);
    out.push(tag);
    if (out.length >= 5) break;
  }
  return out;
}
